compiler/AssociatedTypes.ml

```diff
@@ -238,12 +239,14 @@ let rec norm_ctx_normalize_ty (ctx : norm_ctx) (ty : ty) : ty =
       let trait_ref = norm_ctx_normalize_trait_ref ctx trait_ref in
       let generics = norm_ctx_normalize_generic_args ctx generics in
       (*  For now, we don't support higher order types *)
-      assert (generics = empty_generic_args);
+      let meta = (TraitDeclId.Map.find trait_ref.trait_decl_ref.trait_decl_id ctx.trait_decls).meta in
```

Comment:
You shouldn't use this meta information : the meta-information should be a parameter of the function `norm_ctx_normalize_ty` (the simplest way is probably simply to add a field `meta` in`norm_ctx`). Then, you should add a parameter `meta` to functions like `mk_norm_ctx`, `ctx_normalize_ty`, etc. below.
Those functions will finally be called in different places where you can get the relevant meta-information. For instance, we use those functions to normalize the types used as parameters to a function in a call statement.